frame/evm/precompile/contracts/transfer/src/kton.rs

```diff
@@ -83,25 +65,20 @@ impl<T: Config + dvm_ethereum::Config> Precompile for Kton<T> {
 				);
 				// Ensure caller's balance is enough
 				ensure!(
-					T::KtonAccountBasic::account_basic(&context.caller).balance >= call_data.value,
+					T::KtonAccountBasic::account_basic(&caller).balance >= value,
 					ExitError::OutOfFund
 				);
 
 				// Transfer kton from sender to KTON wrapped contract
-				T::KtonAccountBasic::transfer(
-					&context.caller,
-					&call_data.wkton_address,
-					call_data.value,
-				)?;
+				T::KtonAccountBasic::transfer(&caller, &wkton, value)?;
 				// Call WKTON wrapped contract deposit
-				let precompile_address = H160::from_str(KTON_PRECOMPILE).unwrap_or_default();
-				let raw_input = make_call_data(context.caller, call_data.value)?;
+				let raw_input = make_call_data(caller, value)?;
 				if let Ok(call_res) = T::Runner::call(
-					precompile_address,
-					call_data.wkton_address,
+					array_bytes::hex_into_unchecked(TRANSFER_ADDR),
```

Comment:
We need to be carefully when using `array_bytes::hex_into_unchecked` in runtime, since it is using `unwrap()` under the ground, which could break security audit rules.

But since `TRANSFER_ADDR` here is const and I think we are good. But reviewer should always be nervous on this usage.

Reply:
I suggest to change to use `hex2bytes` here.

```
/// assert_eq!(
/// 	array_bytes::hex2bytes("0x4c6f7665204a616e6520466f7265766572").unwrap_or_default(),
/// 	*b"Love Jane Forever"
/// );
```